drivers/led_strip/tlc5971.c

```diff
@@ -0,0 +1,329 @@
+ */
+#define TLC5971_WRITE_COMMAND (0x25)
+
+/**
+ * @brief GS reference clock select bit in FC data (0 = internal oscillator clock, 1 = SCKI clock).
+ */
+#define TLC5971_CTRL_BIT_EXTGCK (BIT(0))
+
+/**
+ * @brief GS reference clock edge select bit for OUTXn on-off timing control in FC data
+ */
+#define TLC5971_CTRL_BIT_OUTTMG (BIT(1))
+
+/**
+ * @brief Constant-current output enable bit in FC data (0 = output control enabled, 1 = blank).
+ */
+#define TLC5971_CTRL_BIT_BLANK (BIT(5))
+
+/**
+ * @brief Auto display repeat mode enable bit in FC data (0 = disabled, 1 = enabled).
+ */
+#define TLC5971_CTRL_BIT_DSPRPT (BIT(6))
+
+/**
+ * @brief Display timing reset mode enable bit in FC data (0 = disabled, 1 = enabled).
+ */
+#define TLC5971_CTRL_BIT_TMGRST (BIT(7))
+
+/**
+ * @brief Bit mask for control bits in data byte 27
+ */
+#define TLC5971_BYTE27_CTRL_MASK (TLC5971_CTRL_BIT_EXTGCK | TLC5971_CTRL_BIT_OUTTMG)
+
+/**
+ * @brief Bit mask for control bits in data byte 26
+ */
+#define TLC5971_BYTE26_CTRL_MASK                                                                   \
+	(TLC5971_CTRL_BIT_BLANK | TLC5971_CTRL_BIT_DSPRPT | TLC5971_CTRL_BIT_TMGRST)
+
+/**
+ * @brief create data byte 27 from control data
+ *
+ * @param control_data control bits
+ * @return uint8_t the serialized data byte 27
+ */
+static inline uint8_t tlc5971_data_byte27(uint8_t control_data)
+{
+	return (TLC5971_WRITE_COMMAND << 2) | (control_data & TLC5971_BYTE27_CTRL_MASK);
+}
+
+/**
+ * @brief create data byte 26 from control data and blue GBC
+ *
+ * @param control_data control bits
+ * @param gbc_blue global brightness control for blue LEDs
+ * @return uint8_t the serialized data byte 26
+ */
+static inline uint8_t tlc5971_data_byte26(uint8_t control_data, uint8_t gbc_blue)
+{
+	return (control_data & TLC5971_BYTE26_CTRL_MASK) | (gbc_blue >> 2);
+}
+
+/**
+ * @brief create data byte 25 from blue and green GBC
+ *
+ * @param gbc_blue global brightness control for blue LEDs
+ * @param gbc_green global brightness control for green LEDs
+ * @return uint8_t the serialized data byte 25
+ */
+static inline uint8_t tlc5971_data_byte25(uint8_t gbc_blue, uint8_t gbc_green)
+{
+	return (gbc_blue << 6) | (gbc_green >> 1);
+}
+
+/**
+ * @brief create data byte 24 from green and red GBC
+ *
+ * @param gbc_green global brightness control for green LEDs
+ * @param gbc_red global brightness control for red LEDs
+ * @return uint8_t the serialized data byte 24
+ */
+static inline uint8_t tlc5971_data_byte24(uint8_t gbc_green, uint8_t gbc_red)
+{
+	return (gbc_green << 7) | gbc_red;
+}
+
+/**
+ * @brief SPI operation word constant, SPI mode 0, CPOL = 0, CPHA = 0
+ */
+static const uint16_t SPI_OPERATION = (SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | SPI_WORD_SET(8));
+
+static const struct tlc5971_config *get_device_config(const struct device *dev)
+{
+	return dev->config;
+}
+
+static struct tlc5971_data *get_device_data(const struct device *dev)
+{
+	return dev->data;
+}
```

Comment:
This style of accessing device config and data structures is now deprecated. Please use this style instead in any function that needs to access them:

```c

	const struct tlc5971_config *cfg = dev->config;
	struct tlc5971_data *data = dev->data;
```

C will take care of the implicit cast from void*.